applications/OptimizationApplication/python_scripts/algorithms/standardized_rgp_constraint.py

```diff
@@ -0,0 +1,267 @@
+        "<=,
+        ">",
+        ">="
+
+    The reference value for the constraint either can be the "initial_value" or a specified value.
+
+    It can be used only with relaxed_gradient_projection_algorithm. The class contains neccessary functions to compute critical zones. 
+
+    """
+    def __init__(self, parameters: Kratos.Parameters, master_control: MasterControl, optimization_problem: OptimizationProblem, required_buffer_size: int = 4):
+        default_parameters = Kratos.Parameters("""{
+            "response_name"       : "",
+            "type"                : "",
+            "scaled_ref_value"    : 0.0,
+            "buffer_factor"       : 2.0,
+            "max_w_c"             : 10.0,
+            "tolerance"           : 1e-3
+        }""")
+
+        if parameters.Has("scaled_ref_value") and parameters["scaled_ref_value"].IsDouble():
+            default_parameters["scaled_ref_value"].SetDouble(0.0)
+        elif parameters.Has("scaled_ref_value") and parameters["scaled_ref_value"].GetString() == "initial_value":
+            default_parameters["scaled_ref_value"].SetString("initial_value")
+        else:
+            type = parameters["scaled_ref_value"].GetString()
+            raise RuntimeError(f"Provided \"scaled_ref_value\" = {type} is not supported for constraint response functions. Followings are supported options: \n\tinitial_value\n\tfloat value")
+
+        parameters.ValidateAndAssignDefaults(default_parameters)
+
+        response = optimization_problem.GetResponse(parameters["response_name"].GetString())
+
+        super().__init__(master_control, response)
+
+        if required_buffer_size < 4:
+            raise RuntimeError(f"Standardized RGP constraint requires 4 as minimum buffer size. [ response name = {self.GetResponseName()} ]")
+
+        component_data_view = ComponentDataView(response, optimization_problem)
+        component_data_view.SetDataBuffer(required_buffer_size)
+
+        self.__optimization_problem = optimization_problem
+        self.__buffered_data = component_data_view.GetBufferedData()
+        self.__unbuffered_data = component_data_view.GetUnBufferedData()
+
+        if parameters["scaled_ref_value"].IsDouble():
+            self.__reference_value = parameters["scaled_ref_value"].GetDouble()
+        elif parameters["scaled_ref_value"].IsString() and parameters["scaled_ref_value"].GetString() == "initial_value":
+            self.__reference_value = None
+
+        self.__constraint_type = parameters["type"].GetString()
+        if self.__constraint_type in ["<=", "="]:
+            self.__scaling = 1.0
+        elif self.__constraint_type in [">="]:
+            self.__scaling = -1.0
+            if not self.__reference_value is None: self.__reference_value *= -1
+        else:
+            raise RuntimeError(f"Provided \"type\" = {self.__constraint_type} is not supported in constraint response functions. Followings are supported options: \n\t=\n\t<=\n\t<\n\t>=\n\t>")
```

Comment:
consider using an `enum.Enum`:
```py
class ConstraintType(Enum):
    EQUAL = "="
    # ...

if self.constaint_type == Constraint.EQUAL:
    # ....
```

You get the idea, this way it is type safe an easier to read

Reply:
works nice, thnx